File: 剑指offer/57.2.和为s的连续正数序列.py
from typing import List


class Solution:
    def findContinuousSequence(self, target: int) -> List[List[int]]:
        right = 1
        left = 1
        sum_ = 0
        ans = []
        while left < target // 2:
            if sum_ > target:
                sum_ -= left
                left += 1
            elif sum_ < target:
                sum_ += right
                right += 1
            else:
                ans.append(list(range(left, right)))
                sum_ -= left
                left += 1
        return ans

File: 剑指offer/57.2.和为s的连续正数序列.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> findContinuousSequence(int target)
    {
        int left = 1;
        int right = 1;
        int window = 0;

        vector<vector<int>> ans;

        while (left < target / 2) {
            if (window == target) {
                vector<int> tmp;

                for (int i = left; i < right; i++) {
                    tmp.push_back(i);
                }

                ans.push_back(tmp);

                window -= left;
                left++;
            } else if (window < target) {
                window += right;
                right++;
            } else {
                window -= left;
                left++;
            }
        }

        return ans;
    }
};